log/logrus/logrus_logger.go

```diff
@@ -0,0 +1,31 @@
+package log
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/sirupsen/logrus"
+)
+
+type logrusLogger struct {
+	*logrus.Logger
+}
+
+// NewLogrusLogger takes a *logrus.Logger and returns
+//a logger that stisfies the go-kit log.Logger interface
+func NewLogrusLogger(logger *logrus.Logger) log.Logger {
+	return &logrusLogger{logger}
+}
+
+func (l logrusLogger) Log(keyvals ...interface{}) error {
+	if len(keyvals)%2 == 0 {
+		fields := logrus.Fields{}
+		for i := 0; i < len(keyvals); i += 2 {
+			fields[fmt.Sprint(keyvals[i])] = keyvals[i+1]
```

Comment:
`keyvals[i+1]` will panic if `len(keyvals)` is odd. Use `nil` as the field value instead.

Reply:
I see that the `if len(keyvals)%2 == 2` protects against the panic now, but why does the `else` clause just call `l.Info(keyvals)` in that case?

Reply:
l.Info(keyvals) just prints out the arguments as values without keys.
Eg - 
logger.Log("hello", "world", "hello") gives
time="13-09-2018 18:23:14" level=info msg="[hello world hello]"

Reply:
That approach is not compatible with the `log.Logger` interface defined by Go Kit:

>Logger is the fundamental interface for all log operations. Log creates a log event from keyvals, a variadic sequence of alternating keys and values.

We don't want to encourage people to use it without keys because that will be incompatible with other implementations of the interface.

Please look at how the `jsonLogger` handles this situation here: https://github.com/go-kit/kit/blob/master/log/json_logger.go#L23.